includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -1807,19 +1807,12 @@ private function replace_node_with_children( $node ) {
 			return;
 		}
 
-		/*
-		 * If node has children, replace it with them and push children onto stack
-		 *
-		 * Create a DOM fragment to hold the children
-		 */
-		$fragment = $this->dom->createDocumentFragment();
-
 		// Add all children to fragment/stack.
-		$child = $node->firstChild;
-		while ( $child ) {
-			$fragment->appendChild( $child );
-			$this->stack[] = $child;
```

Comment:
Here, all children were appended to `$fragment`. Even though `$fragment` is only used if `$should_replace`. 

So I think this caused the nodes in `$this->stack` to have no values, and thus not get validated:

<img width="826" alt="this-stack" src="https://user-images.githubusercontent.com/4063887/46923408-e403a900-cfdc-11e8-8c0e-c002fbf47810.png">

As a workaround, this stores the nodes in `$children`, and only appends them to `$fragment` if `$should_replace`.